cpp/include/raft/util/device_loads_stores.cuh

```diff
@@ -148,6 +149,26 @@ DI void sts(int32_t* addr, const int32_t (&x)[4])
                : "l"(s4), "r"(x[0]), "r"(x[1]), "r"(x[2]), "r"(x[3]));
 }
 
+DI void sts(half* addr, const half& x)
+{
+  auto s = __cvta_generic_to_shared(reinterpret_cast<uint16_t*>(addr));
+  asm volatile("st.shared.u16 [%0], {%1};" : : "l"(s), "h"(*reinterpret_cast<const uint16_t*>(&x)));
+}
+DI void sts(half* addr, const half (&x)[1])
+{
+  auto s = __cvta_generic_to_shared(reinterpret_cast<uint16_t*>(addr));
+  asm volatile("st.shared.u16 [%0], {%1};" : : "l"(s), "h"(*reinterpret_cast<const uint16_t*>(x)));
+}
+DI void sts(half* addr, const half (&x)[2])
```

Comment:
We do not need this with larger vector size?

Reply:
I think we need all of these, up two 16-byte-long vectors. That's because of the automatic templated vectorization in the ivf-flat scan kernel.
If I remember correctly, every instance I added here was requested by the compiler complaining for the missing instances.

Reply:
Ah, I misunderstood the question on the first read.
Apparently, we don't need `sts` with `half` data type and vector sizes larger than two at the moment. Though since you both suggest we'd better have these for symmetry, I'll add them.